src/bio/bio_xstream.c

```diff
@@ -1038,47 +1043,100 @@ get_bio_blobstore(struct bio_blobstore *bb, struct bio_xs_context *ctxt)
 	return bb;
 }
 
+unsigned int
+smd_dev_type2bdev_role(enum smd_dev_type st)
+{
+	switch (st) {
+	case SMD_DEV_TYPE_DATA:
+		return NVME_ROLE_DATA;
+	case SMD_DEV_TYPE_META:
+		return NVME_ROLE_META;
+	case SMD_DEV_TYPE_WAL:
+		return NVME_ROLE_WAL;
+	default:
+		D_ASSERT(0);
+		return NVME_ROLE_DATA;
+	}
+}
+
+static bool
+is_bdev_match(struct bio_bdev *d_bdev, unsigned int bdev_roles)
+{
+	return d_bdev->bb_roles == 0 || d_bdev->bb_roles & bdev_roles;
```

Comment:
Why bb_roles == 0 means a match?

Reply:
Originally this is for AIO testing purpose, one case is roles might be 0 if metadata on SSD not enabled? maybe we should make roles 0 match DATA instead.